includes/kings-conquest.xml

```diff
@@ -537,78 +615,218 @@
+            <switch-scope inner="player">
+                <set var="king_chance" value="random()"/>
+            </switch-scope>
+            <unless constant="king-race-start" constant-comparison="defined value">
+                <switch-scope inner="team" filter="team_has_king=0">
+                    <action id="crown-random-king"/>
+                </switch-scope>
+            </unless>
         </action>
     </trigger>
-    <trigger filter="always" scope="team">
-        <action>
-            <set var="king_health" value="100"/>
-            <action filter="show-debug-message">
-                <message text="`7» `eKing health set to 100"/>
+    <action id="crown-king" scope="player" filter="all(participating, match_phase=0, team_has_king=0)">
+        <if constant="king-race-start" constant-comparison="defined value">
+            <action filter="only-team-one">
+                <switch-scope inner="match">
+                    <message text="`bThe ${team-one-text-color-symbol}${team-one-name} ${king-noun} `bhas been crowned!"/>
+                </switch-scope>
             </action>
-        </action>
-    </trigger>
-    <trigger filter="reduce-king-health" scope="player">
-        <action>
-            <action filter="only-king">
-                <kit id="king-armor-kit"/>
+            <action filter="only-team-two">
+                <switch-scope inner="match" filter="only-team-two">
+                    <message text="`bThe ${team-two-text-color-symbol}${team-two-name} ${king-noun} `bhas been crowned!"/>
+                </switch-scope>
             </action>
             <switch-scope inner="team">
-                <set var="king_health" value="king_health-1"/>
-                <action filter="show-debug-message">
-                    <message text="`7» `eKing health reduced by 1"/>
+                <sound preset="OBJECTIVE_GOOD"/>
+                <action filter="not(all(team_one_has_king=1, team_two_has_king=1))">
+                    <message text="`7`oThe Conquest will begin once the other team crowns their ${king-noun}."/>
                 </action>
+                <switch-scope inner="player">
+                    <action filter="match-started">
+                        <kit id="race-kit-spectating"/>
+                    </action>
+                </switch-scope>
             </switch-scope>
+            <message text=" "/>
+            <message text="`6» » » `e`lYou are the ${king-noun} `6« « «"/>
+            <message text=" "/>
+        </if>
+        <switch-scope inner="team">
+            <set var="king_health" value="100"/>
+            <set var="team_has_king" value="1"/>
+            <switch-scope inner="player">
+                <set var="is_king" value="0"/>
+                <set var="king_chance" value="99"/>
+            </switch-scope>
+        </switch-scope>
+        <set var="is_king" value="1"/>
+        <set var="king_chance" value="99"/>
+    </action>
+    <action id="crown-random-king" scope="team">

```

Comment:
If this action isnt the one that determines the value of king_chance, you can have late joiners who all have a 0, and are all crowned as king.

Reply:
I haven't been able to create a situation where this is true. Joining after the match starts but within the 30 seconds of this action being called, you still receive a random value for `king_chance`. However, as nothing else except this action uses this variable, it does make more sense to be part of this action.